server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -300,7 +300,7 @@ private void registerRequestHandler(DynamicActionRegistry dynamicActionRegistry)
      * Loads a single extension
      * @param extension The extension to be loaded
      */
-    public void loadExtension(Extension extension) throws IOException {
+    public DiscoveryExtensionNode loadExtension(Extension extension) throws IOException {
```

Comment:
We could make it package private now, right?

```suggestion
    DiscoveryExtensionNode loadExtension(Extension extension) throws IOException {
```

Reply:
We have spies for this in org.opensearch.extensions.rest.RestInitializeExtensionActionTests. I end up with this:

```
        Mockito.doCallRealMethod().when(spy).loadExtension(any(Extension.class));
        Mockito.doNothing().when(spy).initializeExtensionNode(any(DiscoveryExtensionNode.class));
```

We need to stub the second one, but not the first one. So I think it's a bit trickier.